scripts/optimize-assets.sh

```
@@ -0,0 +1,483 @@
+log_warn()    { echo -e "${YELLOW}[SKIP]${NC} $*"; return 0; }
+log_error()   { echo -e "${RED}[ERROR]${NC} $*" >&2; return 0; }
+log_dry()     { echo -e "${YELLOW}[DRY-RUN]${NC} $*"; return 0; }
+
+# ---------------------------------------------------------------------------
+# Argument parsing
+# ---------------------------------------------------------------------------
+
+show_help() {
+    sed -n '2,/^$/p' "$0" | sed 's/^# \?//'
+    exit 0
+    return 0
```

Shellcheck (0.11.0)

[info] Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

(SC2317)